imgee/views/files.py

```diff
@@ -45,7 +35,11 @@ def upload_file(profile):
         if imgfile.filename != '':
             title, stored_file = save_file(imgfile, profile=profile)
             flash('"%s" uploaded successfully.' % title)
-            return redirect(_redirect_url_frm_upload(profile.name))
+            referrer = request.referrer or ''
+            if profile.url_for('pop_up_gallery') in referrer:
```

Comment:
Should be:
1. `profile.url_for(..., _external=True)`
2. urlparse(lhs).netloc+path == urlparse(rhs).netloc+path